addons/purchase/models/account_invoice.py

```diff
@@ -96,6 +98,29 @@ def _onchange_partner_id(self):
                 self.currency_id = self.partner_id.property_purchase_currency_id
         return res
 
+    @api.depends('line_ids.purchase_line_id')
+    def _compute_origin_po_count(self):
+        for move in self:
+            move.purchase_order_count = len(move.line_ids.purchase_line_id.order_id)
+
+    def action_view_source_purchase_orders(self):
+        self.ensure_one()
+        if not self.is_purchase_document():
+            raise UserError(_('You can only view the source Purchase Orders for a Vendor Bill.'))
+        source_orders = self.line_ids.purchase_line_id.order_id
+        if not source_orders:
+            raise UserError(_("There are no source documents."))
```

Comment:
We have already excluded these cases by hiding the button in the view, are you sure you still want to double-check here?

Reply:
I just wanted to be 100% sure but I can remove these